### PR TITLE
Delete a level from a role

### PR DESCRIPTION
**Pull request by silva-00**
Action button to delete a level from a role

### CODE REVIEW

File: backend/src/main/java/saci/domain/model/Level.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package saci.domain.model;
+
+import jakarta.persistence.*;
+import lombok.Data;
+
+@Data
+@Entity
+@Table(name = "levels")
+public class Level {
+    @Id
```
Comment (vitorlofonseca) [inline backend/src/main/java/saci/domain/model/Level.java:10]: These kind of database implementation details should not be here (in the domain). @vascolusitan will explain this to you.

File: backend/src/main/java/saci/domain/service/LevelService.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import saci.domain.model.Level;
+import saci.domain.service.exceptions.AlreadyExistsException;
+import saci.domain.service.exceptions.NotFoundException;
+import saci.infrastructure.LevelRepository;
+
+@Service
+@RequiredArgsConstructor
+public class LevelService {
+
+    private final LevelRepository LevelRepository;
```
Comment (vitorlofonseca) [inline backend/src/main/java/saci/domain/service/LevelService.java:16]:
```suggestion
    private final LevelRepository levelRepository;
```

File: backend/src/main/java/saci/domain/service/LevelService.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        }
+        return LevelRepository.save(Level);
+    }
+
+    public List<Level> getLevels() {
+        return LevelRepository.findAll();
+    }
+
+    public void deleteLevelById(long LevelId) {
+        Optional<Level> Level = LevelRepository.findById(LevelId);
```
Comment (vitorlofonseca) [inline backend/src/main/java/saci/domain/service/LevelService.java:31]:
```suggestion
        Optional<Level> level = LevelRepository.findById(LevelId);
```

Variables should be camelCase as well

File: frontend/src/stores/roles/actions.ts [hunk 4]
```diff
@@ -24,6 +24,11 @@ async function addRole(role: Role): Promise<void> {
   roles.value.push(role)
 }
 
+async function addLevel(level: Level, role: Role): Promise<void> {
```
Comment (vitorlofonseca) [inline frontend/src/stores/roles/actions.ts:27]: Your development is about level removal, not level creation. Please, remove everything not related to levels removal

File: backend/src/main/java/saci/application/LevelController.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+import io.swagger.v3.oas.annotations.media.ArraySchema;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
```
Comment (vascolusitan) [inline backend/src/main/java/saci/application/LevelController.java:13]: Watch out for star imports.

File: backend/src/main/java/saci/domain/service/LevelService.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
+package saci.domain.service;
+
```
Comment (vascolusitan) [inline backend/src/main/java/saci/domain/service/LevelService.java:2]: Don't forget to do the clean up of everything not related to the ticket in the backend, as Vitor said.

File: backend/src/main/java/saci/domain/service/LevelService.java [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+import saci.domain.service.exceptions.NotFoundException;
+import saci.infrastructure.LevelRepository;
+
+@Service
+@RequiredArgsConstructor
+public class LevelService {
+
+    private final LevelRepository levelRepository;
+
+    public List<Level> getLevels() {
```
Comment (vitorlofonseca) [inline backend/src/main/java/saci/domain/service/LevelService.java:17]: You don't need this, right?

File: backend/src/main/java/saci/domain/service/LevelService.java [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+        } else {
+            throw new NotFoundException("Level not found with ID: " + levelId);
+        }
+    }
+
+    public Optional<Level> findById(long levelId) {
+        return levelRepository.findById(levelId);
+    }
+
+    public List<Level> getLevelsByRoleId(Long roleId) {
```
Comment (vitorlofonseca) [inline backend/src/main/java/saci/domain/service/LevelService.java:30]: Is this still necessary?

File: backend/src/main/java/saci/infrastructure/LevelRepository.java [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+
+import java.util.List;
+import java.util.Optional;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+import saci.domain.model.Level;
+
+@Repository
+public interface LevelRepository extends JpaRepository<Level, Long> {
+    Optional<Level> findByName(String name);
```
Comment (vitorlofonseca) [inline backend/src/main/java/saci/infrastructure/LevelRepository.java:11]: And these, necessary?

File: frontend/src/router/index.ts [hunk 10]
```diff
@@ -34,6 +35,10 @@ const router = createRouter({
       path: '/deleteknowledge',
       component: DeleteKnowledgeView
     },
+    {
+      path: '/deletelevel',
```
Comment (vitorlofonseca) [inline frontend/src/router/index.ts:39]:
```suggestion
      path: '/delete-level',
```

A path should be snake-case

**Review by vitorlofonseca — APPROVED** [review]

**Review by fromgonie — APPROVED** [review]
good job!

**Review by vascolusitan — APPROVED** [review]

**Pull request merged**